api/pkg/apis/v1alpha1/providers/states/k8s/k8s.go

```diff
@@ -251,13 +302,68 @@ func (s *K8sStateProvider) Upsert(ctx context.Context, entry states.UpsertReques
 			item.SetNamespace(metadata.Namespace)
 			item.SetLabels(metadata.Labels)
 			item.SetAnnotations(metadata.Annotations)
+
```

Comment:
I still think we need container scope lock, or any other sync mechanism to prevent the concurrent access to container_latest tag.